### ISSUE TITLE
unknown parameter 'model_partial'

### ISSUE THREAD

**Issue by varosi-chaosgroup** [ucsd-progsys/liquidhaskell#2094]
Related to issue https://github.com/ucsd-progsys/lh-plugin-demo/issues/16

Running on macOS 13.0 64bit.
I have z3 --version
Z3 version 4.11.2 - 64 bit
I cloned latest [lh-plugin-demo](https://github.com/ucsd-progsys/lh-plugin-demo) and ran: **stack build**
I got:
```
liquid-prelude> Linking /private/var/folders/9f/s1y16f_95tq8ldsqzc2kfmm80000gn/T/stack-2497240cd673accb/liquid-prelude-0.8.10.2/.stack-work/dist/x86_64-osx/Cabal-3.2.1.0/setup/setup ...
liquid-prelude> Configuring liquid-prelude-0.8.10.2...
liquid-prelude> build
liquid-prelude> Preprocessing library for liquid-prelude-0.8.10.2..
liquid-prelude> Building library for liquid-prelude-0.8.10.2..
liquid-prelude> [ 1 of 11] Compiling Language.Haskell.Liquid.Bag
liquid-prelude> 
liquid-prelude> **** LIQUID: ERROR *************************************************************
liquid-prelude> 
liquid-prelude> <no location info>: error:
liquid-prelude>     /private/var/folders/9f/s1y16f_95tq8ldsqzc2kfmm80000gn/T/stack-2497240cd673accb/liquid-prelude-0.8.10.2:1:1-1:1: Error
liquid-prelude>   crash: SMTLIB2 respSat = Error "line 4 column 27: unknown parameter 'model_partial', this is an old parameter name, invoke 'z3 -p' to obtain the new parameter list"
liquid-prelude> 
liquid-prelude>
```

Which led to failure with the build. I think that it could help if compatible Z3 versions are written in the requirements.

**Comment by facundominguez** [comment]
Thanks @varosi-chaosgroup. I've tried a fix in ucsd-progsys/lh-plugin-demo#17

**Comment by facundominguez** [comment]
Please, reopen if building still fails for you.
